tools/checked-c-convert/ConstraintBuilder.cpp

```diff
@@ -183,130 +197,161 @@ class FunctionVisitor : public RecursiveASTVisitor<FunctionVisitor> {
   // assigning to. V represents constraints on a pointer variable. RHS is 
   // an expression which might produce constraint variables, or, it might 
   // be some expression like NULL, an integer constant or a cast.
-  void constrainAssign( std::set<ConstraintVariable*> V, 
+  void constrainLocalAssign( std::set<ConstraintVariable*> V,
                         QualType lhsType,
                         Expr *RHS) {
     if (!RHS || V.size() == 0)
       return;
 
+    std::set<ConstraintVariable *> RHSConstraints;
+    RHSConstraints.clear();
+
     Constraints &CS = Info.getConstraints();
-    std::set<ConstraintVariable*> W = Info.getVariable(RHS, Context);
-    if (W.size() > 0) {
-      // Case 1.
-      // There are constraint variables for the RHS, so, use those over
-      // anything else we could infer. 
-      constrainEq(V, W, Info);
+    RHS = getNormalizedExpr(RHS);
+    CallExpr *CE = dyn_cast<CallExpr>(RHS);
+    // if this is a call expression to a function.
+    if (CE != nullptr && CE->getDirectCallee() != nullptr) {
+      // case 5
+      // if this is a call expression?
+      // is this functions return type an itype
```

Comment:
I think this would be clearer worded as "Check if this functions' return type is an itype".

Reply:
Fixed.